src/components/DuosHeader.jsx

```diff
@@ -0,0 +1,537 @@
+  const [state, setState] = useState({
+    showSupportRequestModal: false,
+    hover: false,
+    notificationData: [],
+    openDrawer: false
+  });
+
+  useEffect(() => {
+    async function fetchNotificationData() {
+      const notificationData = await NotificationService.getActiveBanners();
+      setState((prev) => ({
+        ...prev,
+        notificationData
+      }));
+    }
+    fetchNotificationData();
+  }, []);
+
+  const toggleHover = () => {
+    setState({
+      ...state,
+      hover: !state.hover
+    });
+  };
+
+  const signOut = () => {
+    props.history.push('/home');
+    toggleDrawer(false);
+    // Assuming `onSignOut` is a prop function passed down
+    // from the parent component
+    props.onSignOut();
+  };
+
+  const supportRequestModal = () => {
+    setState({
+      ...state,
+      showSupportRequestModal: true,
+      openDrawer: false
+    });
+  };
+
+  const okSupportRequestModal = () => {
+    setState({
+      ...state,
+      showSupportRequestModal: false
+    });
+  };
+
+  const closeSupportRequestModal = () => {
+    setState({
+      ...state,
+      showSupportRequestModal: false
+    });
+  };
+
+  const makeNotifications = () => {
+    return state.notificationData.map((d, index) => <Notification notificationData={d} key={index} />);
+  };
+
+  const toggleDrawer = (boolVal) => {
+    setState({
+      ...state,
+      openDrawer: boolVal
+    });
+  };
+
+  const goToLink = (link) => {
+    props.history.push(link);
+    toggleDrawer(false);
+  };
+
+  let isLogged = Storage.userIsLogged();
+  let currentUser = {};
+
+  if (isLogged) {
+    currentUser = Storage.getCurrentUser();
+  }
+
+  const contactUsSource = state.hover ? contactUsHover : contactUsStandard;
+  const contactUsIcon = isLogged ? '' : <img src={contactUsSource} style={{ display: 'inline-block', margin: '0 8px 0 0', verticalAlign: 'baseline' }} />;
+  const contactUsText = isLogged ? 'Contact Us' : <span style={{ display: 'inline', verticalAlign: 'text-bottom' }}>Contact Us</span>;
+  const contactUsButton = (
+    <button
+      id="btn_applyAcces"
+      style={{
+        color: state.hover ? '#2FA4E7' : '#ffffff',
+        fontSize: '14px',
+        fontWeight: '500',
+        background: 'transparent',
+        border: 'none',
+        outline: 'none',
+        margin: '0 20px 0 0',
+        padding: '15px 0',
+      }}
+      onMouseEnter={toggleHover}
+      onMouseLeave={toggleHover}
+      onClick={supportRequestModal}
+      data-tip="Need help? Contact us for some assistance"
+      data-for="tip_requestAccess"
+    >
+      {contactUsIcon}
+      {contactUsText}
+    </button>
+  );
+
+
+  const supportModal = (
+    <SupportRequestModal
+      showModal={state.showSupportRequestModal}
+      onOkRequest={okSupportRequestModal}
+      onCloseRequest={closeSupportRequestModal}
+      url={props.location.pathname}
+    />
+  );
+
+  const tabs = headerTabsConfig.filter((data) => data.isRendered(currentUser));
+
```

Comment:
I think we are missing some comments from the previous version:
```suggestion
  // returns true if the current page the app is on is a part of this tab
```